### DISCUSSION TITLE
Large middleware size on build and webpack notification

### DISCUSSION THREAD

**Discussion by mertafor** [amannn/next-intl#297] [Q&A]
I noticed that middleware size is too large at build due to `createIntlMiddleware` middleware. Middleware size is 55kb by using only nextauth middleware. However, if I also include createIntlMiddleware, it becomes 225kb (displayed as red, apparently too large for middleware file size). 

Also, this middleware function brings this webpack notification in console :
<w> [webpack.cache.PackFileCacheStrategy] Serializing big strings (659kiB) impacts deserialization performance (consider using Buffer instead and decode when needed)

In order to make sure, I performed several tests by adding / removing createIntlMiddleware.

My question is, does this library require using this middleware? For now, I removed it and using redirection solution which is provided in NextJS docs (redirect if language code is missing in url path)

**Comment by amannn** [comment]
This is true, thanks for the report! There was a regression in Next.js 13.3 that caused this (https://github.com/vercel/next.js/issues/48469). There's theoretically a workaround for libraries, but there was movement on the issue just two days ago, I hope that this gets fixed in a patch release of Next.js. As a temporary solution, you can downgrade Next.js to <13, if that's an option for you.

**Reply by mertafor** [reply]
Glad to know that there will be a patch coming. Unfortunately I have to use next 13+ since I'm using app directory. Hope the next release fix it. Thanks!
